src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -44,6 +52,7 @@ public override void Reset()
         {
             base.Reset();
             _storages.Reset();
+            _cellCache.Clear();
```

Comment:
Do you only want to clear it if the Keccak changes from last commit to new block?

Reply:
In practice this will be called every block.

Reply:
From offline discussion cache is valid if when new block is started the `StateRoot` == `LastStateRoot`. i.e. the block being worked on is the next block. If its different (due to reorg) then the cache is invalid and should be cleared (very coarse grained cache invalidation)

Reply:
Is this the case though? AFAIK, the `Reset` is called only on a rerog or branch restoration

https://github.com/NethermindEth/nethermind/blob/7bb9dec1f433a884642a2b63212b2112f01a3339/src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs#L157-L165 and https://github.com/NethermindEth/nethermind/blob/7bb9dec1f433a884642a2b63212b2112f01a3339/src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs#L186-L189

This should not be happening on every block then?

Reply:
> From offline discussion cache is valid if when new block is started the StateRoot == LastStateRoot. i.e. the block being worked on is the next block. If its different (due to reorg) then the cache is invalid and should be cleared (very coarse grained cache invalidation)

Wait a sec @benaadams @LukaszRozmej If this is the case, and I think in the comment above I can confirm that only a on a reorg or dropping the head the storage provider is reset, then maybe we could skip the check and just process as is, delegating the reset of the internall cell cache to the `_storageProvider.Reset` with no checks needed?